sentry/interceptor.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import asdict, is_dataclass
+from typing import Any, Optional, Type
+
+import sentry_sdk
+from temporalio import workflow
+from temporalio.worker import (
+    ActivityInboundInterceptor,
+    ExecuteActivityInput,
+    ExecuteWorkflowInput,
+    Interceptor,
+    WorkflowInboundInterceptor,
+    WorkflowInterceptorClassInput,
+)
+
+
+class _SentryActivityInboundInterceptor(ActivityInboundInterceptor):
+    async def execute_activity(self, input: ExecuteActivityInput) -> Any:
+        transaction_name = input.fn.__module__ + "." + input.fn.__name__
```

Comment:
```suggestion
        transaction_name = input.fn.__module__ + "." + input.fn.__qualname__
```

Sometimes activities are methods of classes. Then again, you could use `temporalio.activity.info().activity_type` which is the registered name with the worker.